e2e/test/app.ts

```diff
@@ -0,0 +1,194 @@
+                        .expect(Validate.Apps.checkFor(appName, Validate.doesNotExistInContainer))
+                        .run(Command.appLs())
+                        .end(removeAndDone(done, appName));
+                });
+        });
+        
+        it("fails with n", (done: MochaDone) => {
+            nixt()
+                .stdout("App removal cancelled.")
+                .run(Command.appRm(appName))
+                .on(Command.PROMPT_ARE_YOU_SURE).respond(Command.RESPONSE_REJECT)
+                .end(() => {
+                    nixt()
+                        .expect(Validate.Apps.checkFor(appName, Validate.existsInContainer))
+                        .run(Command.appLs())
+                        .end(done);
+                });
+        });
+        
+        it("fails if app not found", (done: MochaDone) => {
+            var fakeAppName: string = "not_a_real_app";
+            
+            nixt()
+                .stderr(Error.appNotFound(fakeAppName))
+                .run(Command.appRm(fakeAppName))
+                .on(Command.PROMPT_ARE_YOU_SURE).respond(Command.RESPONSE_ACCEPT)
+                .end(done);
+        });
+    });
+    
+    describe("app rename", () => {
+        var oldAppName: string;
+        var newAppName: string;
+        beforeEach((done: MochaDone) => {
+            oldAppName = makeRandomString();
+            newAppName = makeRandomString();
+            
+            nixt()
+                .stdout(Success.appAdd(oldAppName))
+                .run(Command.appAdd(oldAppName))
+                .end(addAndDone(done, oldAppName));
+        });
+        
+        it("succeeds", (done: MochaDone) => {
+            nixt()
+                .stdout(Success.appRename(oldAppName, newAppName))
+                .run(Command.appRename(oldAppName, newAppName))
+                .end(() => {
+                    nixt()
+                        .expect(Validate.Apps.checkForMany({ [oldAppName]: Validate.doesNotExistInContainer, [newAppName]: Validate.existsInContainer }))
+                        .run(Command.appLs())
+                        .end(renameAndDone(done, oldAppName, newAppName));
+                });
+        });
+        
+        it("fails if app not found", (done: MochaDone) => {
+            var fakeAppName: string = "not_a_real_app";
+            
+            nixt()
+                .stderr(Error.appNotFound(fakeAppName))
+                .run(Command.appRename(fakeAppName, newAppName))
+                .end(done);
+        });
+        
+        it("fails if app conflict", (done: MochaDone) => {
+            nixt()
+                .code(201)
+                .run(Command.appAdd(newAppName))
+                .end(() => {
+                    nixt()
+                        .stderr(Error.appConflict(newAppName))
+                        .run(Command.appRename(oldAppName, newAppName))
+                        .end(addAndDone(done, newAppName));
+                });
+        });
+    });
```

Comment:
we also have `app list` `app remove` commands. Although they are the same as `ls` and `rm` respectively, we should add tests for them too.